watertap/unit_models/tests/test_pressure_changer.py

```diff
@@ -259,3 +262,108 @@ def test_solution(self, Pump_frame):
         )
+
+
+class TestPumpVariable_Flow(TestPumpIsothermal):
+    @pytest.fixture(scope="class")
+    def Pump_frame(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.properties = props.SeawaterParameterBlock()
+
+        m.fs.unit = Pump(
+            default={
+                "property_package": m.fs.properties,
+                "variable_efficiency": VariableEfficiency.flow,
+            }
+        )
+        # fully specify system
+        feed_flow_mass = 1
+        feed_mass_frac_TDS = 0.035
+        feed_pressure_in = 1e5
+        feed_pressure_out = 5e5
+        feed_temperature = 273.15 + 25
+        efi_pump = 0.75
+
+        feed_mass_frac_H2O = 1 - feed_mass_frac_TDS
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "TDS"].fix(
+            feed_flow_mass * feed_mass_frac_TDS
+        )
+        m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "H2O"].fix(
+            feed_flow_mass * feed_mass_frac_H2O
+        )
+        m.fs.unit.inlet.pressure[0].fix(feed_pressure_in)
+        m.fs.unit.inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.outlet.pressure[0].fix(feed_pressure_out)
+
+        m.fs.unit.bep_eta.fix(efi_pump)
+
+        flow_vol = m.fs.unit.control_volume.properties_in[0].flow_vol.expr()
+        m.fs.unit.bep_flow.fix(flow_vol)
+        return m
+
+    @pytest.mark.unit
+    def test_build(self, Pump_frame):
+        m = Pump_frame
+
+        # check that IDAES pump model has not changed variable and constraint names
+        var_list = [
+            "bep_flow",
+            "bep_eta",
+            "flow_ratio",
+        ]
+        for v in var_list:
+            assert hasattr(m.fs.unit, v)
+            var = getattr(m.fs.unit, v)
+            assert isinstance(var, Var)
+
+        assert isinstance(m.fs.unit.flow_ratio_constraint, Constraint)
+
+    @pytest.mark.component
+    def test_calculate_scaling(self, Pump_frame):
+        m = Pump_frame
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1, index=("Liq", "H2O")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mass_phase_comp", 1e2, index=("Liq", "TDS")
+        )
+
+        calculate_scaling_factors(m)
+
+        # if IDAES hasn't specified a scaling factor
+        if get_scaling_factor(m.fs.unit.ratioP) is None:
+            set_scaling_factor(m.fs.unit.ratioP, 1)
+
+        if get_scaling_factor(m.fs.unit.control_volume.work) is None:
+            set_scaling_factor(m.fs.unit.control_volume.work, 1)
+
+        # check that all variables and constraints have scaling factors
+        unscaled_var_list = list(unscaled_variables_generator(m))
+        assert len(unscaled_var_list) == 0
+
+        unscaled_constraint_list = list(unscaled_constraints_generator(m))
+        assert len(unscaled_constraint_list) == 0
+
+    @pytest.mark.component
+    def test_solution(self, Pump_frame):
```

Comment:
Should there be a test for the initialization method before this?

It would also be good to have a test for unit consistency as well.